### PR TITLE
feat: Capped product user journey

### PR DESCRIPTION
**Pull request by alainajaffer**
## Description

Creation of new feature flagged user journey called 'Capped products'. 

## Testing instructions

Create a Capped Product from faretypes page onward should be similar to the period journey

### CODE REVIEW

File: repos/fdbt-site/src/pages/api/selectSalesOfferPackage.ts [hunk 1]
```diff
@@ -31,7 +31,6 @@ export const sanitiseReqBody = (
     products: { productName: string }[],
 ): SanitisedBodyAndErrors => {
     const errors: ErrorInfo[] = [];
-
```
Comment (NitishAroraKPMG) [inline repos/fdbt-site/src/pages/api/selectSalesOfferPackage.ts:34]: remove this change please

File: repos/fdbt-site/src/pages/api/salesConfirmation.ts [hunk 2]
```diff
@@ -49,7 +49,12 @@ export default async (req: NextApiRequestWithSession, res: NextApiResponse): Pro
             userDataJson = getSingleTicketJson(req, res);
         } else if (fareType === 'return') {
             userDataJson = getReturnTicketJson(req, res);
-        } else if (fareType === 'period' || fareType === 'multiOperator' || fareType === 'flatFare') {
+        } else if (
+            fareType === 'period' ||
```
Comment (NitishAroraKPMG) [inline repos/fdbt-site/src/pages/api/salesConfirmation.ts:53]: can we combine and use in operator like

fareType in ['period', ...]

Comment (NitishAroraKPMG) [reply repos/fdbt-site/src/pages/api/salesConfirmation.ts:53]: ['period', 'multiOperator', 'flatFare', 'cappedProduct'].includes(fareType)

Comment (alainajaffer) [reply repos/fdbt-site/src/pages/api/salesConfirmation.ts:53]: probably will do this now

File: repos/fdbt-site/src/constants/index.ts [hunk 3]
```diff
@@ -82,7 +82,15 @@ export const INTERNAL_NOC = 'IWBusCo';
...
 
-export const validFareTypes = ['single', 'period', 'return', 'flatFare', 'multiOperator', 'schoolService'];
+export const validFareTypes = [
+    'single',
+    'period',
+    'return',
+    'flatFare',
+    'multiOperator',
+    'schoolService',
+    'cappedProduct',
```
Comment (RobCattonKPMG) [inline repos/fdbt-site/src/constants/index.ts:92]: I don't think we should be adding this here - capped products can be single or return or period etc.

File: repos/fdbt-site/src/interfaces/matchingJsonTypes.ts [hunk 4]
```diff
@@ -74,7 +74,14 @@ export interface SalesOfferPackage {
...
 
-export type TicketType = 'flatFare' | 'period' | 'multiOperator' | 'schoolService' | 'single' | 'return';
+export type TicketType =
+    | 'flatFare'
+    | 'period'
+    | 'multiOperator'
+    | 'schoolService'
+    | 'single'
+    | 'return'
+    | 'cappedProduct';
```
Comment (RobCattonKPMG) [inline repos/fdbt-site/src/interfaces/matchingJsonTypes.ts:84]: Remove from here also please.

File: repos/fdbt-site/src/pages/api/csvZoneUpload.ts [hunk 5]
```diff
@@ -199,7 +199,7 @@ export default async (req: NextApiRequestWithSession, res: NextApiResponse): Pro
                 return;
             }
 
-            if (fareType === 'period') {
+            if (fareType === 'period' || fareType === 'cappedProduct') {
```
Comment (RobCattonKPMG) [inline repos/fdbt-site/src/pages/api/csvZoneUpload.ts:202]: and remove from here also.

File: repos/fdbt-site/src/pages/api/fareConfirmation.ts [hunk 6]
```diff
@@ -7,7 +7,7 @@ import { FARE_TYPE_ATTRIBUTE } from '../../constants/attributes';
 export default (req: NextApiRequestWithSession, res: NextApiResponse): void => {
     try {
         const { fareType } = getSessionAttribute(req, FARE_TYPE_ATTRIBUTE) as FareType;
-        if (isSchemeOperator(req, res) && fareType !== 'period') {
+        if (isSchemeOperator(req, res) && fareType !== 'period' && fareType !== 'cappedProduct') {
```
Comment (RobCattonKPMG) [inline repos/fdbt-site/src/pages/api/fareConfirmation.ts:10]: As above.

File: repos/fdbt-site/src/pages/fareType.tsx [hunk 7]
```diff
@@ -95,6 +95,11 @@ const buildRadioProps = (schemeOp: boolean): RadioOption[] => {
             label: 'Academic term/year ticket',
             hint: 'A ticket available to pupils in full-time education',
         },
+        {
+            value: 'cappedProduct',
+            label: 'Capped product',
+            hint: 'A cap applicable over a number of days or weeks',
```
Comment (RobCattonKPMG) [inline repos/fdbt-site/src/pages/fareType.tsx:101]: Do we not need something in fareType.ts to handle this value?

**Review by RobCattonKPMG — APPROVED** [review]

**Pull request merged**